### PR TITLE
*: style and formatting changes

### PR DESCRIPTION
**Pull request by nrc**
A number of minor and mechanical changes to code style. Some comments:

* everything changed in the PR is enforced mechanically by the final commit
* there were some missing copyright/license headers that were added
* I changed all uses of `TODO` or `todo!` to `FIXME` or `unimplemented!`, this means that devs can use `TODO` or `todo!` for temporary todos and CI ensures they don't forget to address them before committing.

PTAL @Connor1996 @innerr 

### Release note 

* No release note

### CODE REVIEW

**Review by Connor1996 — COMMENTED** [review]
Can we keep the `TODO`s?  `FIXME` is kind of like there is a bug to solve.

**Comment by nrc** [comment]
> Can we keep the TODOs? FIXME is kind of like there is a bug to solve.

I don't really mind what labels we use as long as there is a convention to distinguish "this is todo and must be fixed before I commit" and "this is todo and it can be fixed later". The only convention I know from elsewhere is TODO/FIXME from Rust, but anything that is consistent is fine (there is a slight advantage to use `TODO` for the first meaning, because it corresponds with the `todo!` macro, c.f., `unimplemented!` for longer term stuff. We could use `UNIMPLEMENTED` instead of `FIXME` but it is a bit verbose).

File: components/tidb_query_expr/src/impl_encryption.rs [hunk 1]
```diff
@@ -316,16 +316,56 @@ mod tests {
...
-            ("pingcap", 256, "2871823be240f8ecd1d72f24c99eaa2e58af18b4b8ba99a4fc2823ba5c43930a"),
-            ("pingcap", 384, "c50955b6b0c7b9919740d956849eedcb0f0f90bf8a34e8c1f4e071e3773f53bd6f8f16c04425ff728bed04de1b63db51"),
-            ("pingcap", 512, "ea903c574370774c4844a83b7122105a106e04211673810e1baae7c2ae7aba2cf07465e02f6c413126111ef74a417232683ce7ba210052e63c15fc82204aad80"),
-            ("13572468", 0, "1c91ab1c162fd0cae60a5bb9880f3e7d5a133a65b6057a644b26973d9c55dcfe"),
-            ("13572468", 224, "8ad67735bbf49576219f364f4640d595357a440358d15bf6815a16e4"),
-            ("13572468", 256, "1c91ab1c162fd0cae60a5bb9880f3e7d5a133a65b6057a644b26973d9c55dcfe"),
-            ("13572468.123", 384, "3b4ee302435dc1e15251efd9f3982b1ca6fe4ac778d3260b7bbf3bea613849677eda830239420e448e4c6dc7c2649d89"),
-            ("13572468.123", 512, "4820aa3f2760836557dc1f2d44a0ba7596333fdb60c8a1909481862f4ab0921c00abb23d57b7e67a970363cc3fcb78b25b6a0d45cdcac0e87aa0c96bc51f7f96"),
+            (
+                "pingcap",
```
[the remark this answers is not in the document]
Comment (nrc) [reply components/tidb_query_expr/src/impl_encryption.rs:320]: I do not. I have `#[rustfmt::skip]`ed it

**Comment by nrc** [comment]
> I agree FIXME seems more urgent than TODO. How about using FIXME/unimplemented! as short term task and TODO/todo! for future plans?

This doesn't work because clippy/rustfmt only lint for TODO/todo!, so although we can use what we like for longer term, we have to use those for the short term

**Comment by BusyJay** [comment]
> > I agree FIXME seems more urgent than TODO. How about using FIXME/unimplemented! as short term task and TODO/todo! for future plans?
> 
> This doesn't work because clippy/rustfmt only lint for TODO/todo!, so although we can use what we like for longer term, we have to use those for the short term

If those lints are really required, how about using a standalone script and skip the check in rustfmt (until keyword can be configured)?

**Comment by nrc** [comment]
> If those lints are really required, how about using a standalone script and skip the check in rustfmt (until keyword can be configured)?

We could, but then we are creating our own conventions, rather than following existing conventions. That makes it harder for people to start contributing to our project and means that developers have to expend energy context switching between our conventions and the more widely used ones.

**Comment by BusyJay** [comment]
When searching github, there are 245 million lines of code contains `todo` and 110 million lines of code contains `fixme`. It probably shows that todo is widely accepted as for long term tasks in some level. I suggest to disable the lint for now and see if rustfmt can support custom todo key word.

**Comment by nrc** [comment]
> When searching github, there are 245 million lines of code contains todo and 110 million lines of code contains fixme. It probably shows that todo is widely accepted as for long term tasks in some level. I suggest to disable the lint for now and see if rustfmt can support custom todo key word.

I don't think this a great analysis. It conflates the questions of whether there should be a distinction between 'todo' and something else, and what 'something else' should be. It also doesn't account for the quality of code in a project - there are lots of side-projects, throwaway projects, and projects with just low code quality on GH, and I do not want to compare TiKV with them (and I would bet that such projects have more 'todo' in them than high quality projects do).

In any case, I don't think it is worth too much time worrying about, so I've removed the TODO -> FIXME changes and accompanying change to the rustfmt config. (I've left the change from todo! to unimplemented!, since that is much smaller and hopefully less controversial since the semantics as I understand them are specified in the docs).

PTAL @Connor1996 @BusyJay

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by BusyJay — APPROVED** [review]
Rest LGTM

**Pull request merged**